frontend/src/components/Forms/RegisterPage/RegisterPage.tsx

```diff
@@ -0,0 +1,167 @@
+  const [passwordValid, setPasswordValid] = useState(true);
+  const confirmPasswordField = useRef<HTMLInputElement>(null);
+  const usernameField = useRef<HTMLInputElement>(null);
+  const passwordField = useRef<HTMLInputElement>(null);
+  const [isProcessing, setIsProcessing] = useState(false);
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setFormData({
+      ...formData,
+      [e.target.name]: e.target.value,
+    });
+
+    if (!passwordMatch) {
+      setPasswordMatch(true);
+    }
+    if (!usernameAvaliable) {
+      setUsernameAvaliable(true);
+    }
+    if (usernameEmpty) setUsernameEmpty(false);
+    if (passwordEmpty) setPasswordEmpty(false);
+    if (confirmPasswordEmpty) setConfirmPasswordEmpty(false);
+    if (!passwordValid) setPasswordValid(true);
+
+    usernameField.current?.classList.remove(classes.inputError);
+    passwordField.current?.classList.remove(classes.inputError);
+    confirmPasswordField.current?.classList.remove(classes.inputError);
+  };
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    if (isProcessing) return;
+
+    setIsProcessing(true);
+
+    if (formData.username === "") {
+      usernameField.current?.classList.add(classes.inputError);
+      return setUsernameEmpty(formData.username === "");
+    }
+    if (formData.password === "") {
+      passwordField.current?.classList.add(classes.inputError);
+      return setPasswordEmpty(formData.password === "");
+    }
+    if (formData.confirmPassword === "") {
+      confirmPasswordField.current?.classList.add(classes.inputError);
+      return setConfirmPasswordEmpty(formData.confirmPassword === "");
+    }
+    if (formData.confirmPassword !== formData.password) {
+      confirmPasswordField.current?.classList.add(classes.inputError);
+      return setPasswordMatch(false);
+    }
+
+    if (!passwordPattern.test(formData.password)) {
+      passwordField.current?.classList.add(classes.inputError);
+      confirmPasswordField.current?.classList.add(classes.inputError);
+      return setPasswordValid(false);
+    }
+    try {
+      const resp = await fetch("http://localhost:3000/register", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify(formData),
+      });
+
+      if (resp.ok) {
+        console.log("success");
+        onClick();
+      } else {
+        console.log("failure");
+      }
+    } catch (err) {
+      console.log("Error: ", err);
+    } finally {
+      setIsProcessing(false);
+    }
+  };
+
+  return (
+    <div className={classes.container}>
+      <div className={classes.sheet}>
+        <button className={classes.close} onClick={onClick}>
+          <X />
+        </button>
+        <h1 className={classes.title}>Register!</h1>
+        <form className={classes.form} onSubmit={handleSubmit}>
+          <span className={classes.label}>Username</span>
```

Comment:
i shouldve brought this up earlier in the review but there's a dedicated `<label>` tag you can use that does the same thing. but if you add the `for=` attribute to it, setting its value to the `name` of an `input`, you can then click on the label and it will focus on the input for you. handy QOL thing and its also better for accessibility